### ISSUE TITLE
Misunderstanding how annotations are encoded

### ISSUE THREAD

**Issue by Simon-Bertrand** [smhassanerfani/atlantis#17]
Hello dear,
Firstly thank you for this dataset release, it is very useful for research work.

I have some trouble to understand how labels are encoded using the [labels_info.json](https://github.com/smhassanerfani/atlantis/blob/master/utils/labels_info.json) file.

To start, the labels seem to be shifted by +1. When I open a mask, for example the lighthouses are encoded with the value 22 (should be 21), the sky 42 (should be 41), the sea 38 (should be 37), the vegetation 51 (should be 50). 

![image](https://user-images.githubusercontent.com/84455908/223753749-fab90d31-86c4-4c40-90e3-57a80ef551d1.png)

Can you confirm, please, this choice ? It also seem that this choice is confirmed in this file [dataloader.py](https://github.com/smhassanerfani/atlantis/blob/master/dataloader.py), line 71, but I see no official statements confirming it in the different text explanations.

In [dataloader.py](https://github.com/smhassanerfani/atlantis/blob/master/dataloader.py), line 72, you replace it by 255, which is not a value contained in the [labels_info.json](https://github.com/smhassanerfani/atlantis/blob/master/utils/labels_info.json) file.

The last issue that I have is that it seems that a lot of masks have randomly the value 0 contained in it (Discovered this by using np.unique on few masks). Is this just annotation noise that can lead to considering a supplementary class (like a class to ignore) ? How did you manage this ?

Thank you very much for reading and, in advance, for answering.

**Comment by smhassanerfani** [comment]
Hi Simon,
Let's say labels are shifted by -1 from whatever the label code is in the mask compared with the JSON file. The reason is you don't need to train your models for background (0 value in the masks). So what we did (as you noticed in line 71 of `dataloader.py`) we shifted all labels to -1 and then considered -1 as 255 to be excluded in the evaluation (`train.py` line 128: `loss_fn = torch.nn.CrossEntropyLoss(ignore_index=255)`. 
In the mask zero value is the background label so it's so normal you find some values sporadically. still, they are not a lot as in CVAT we did our best to use overlap and minimize such situations.

Good luck!
